### PR TITLE
SC-132

### PR DESCRIPTION
**Pull request by xschildw**
Proposal: 
- shared.xml to be run once per account, creates ssl cert for friendlyname *.sageit.org
- app-extra.xml to create 1. add friendly ssl cert created above to https listener, 2. redirect http to https on actual endpoint, 3. redirect requests to friendlyname to actual endpoint

Notes:
. reference to shared stack when importing friendlyname ssl cert. Implies shared stack name. Good?
. reference to app endpoint could be changed to import app stack endpoint but would have to pass stack name as param, seemed clearer to just pass the endpoint.

### CODE REVIEW

File: templates/app-extra.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          Query: '#{query}'
+          Path: '/#{path}'
+          Port: '443'
+          StatusCode: HTTP_302
+      Conditions:
+      - Field: host-header
+        Values:
+        - !Ref FriendlyEndpoint
+      ListenerArn: !Ref Listener443Arn
+      Priority: 1
```
Comment (zaro0508) [inline templates/app-extra.yaml:68]: should this priority be different than HttpRedirectRule?

File: templates/app-extra.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    Description: Host name to redirect to
+    Type: String
+  FriendlyEndpoint:
+    Description: Friendly endpoint
+    Type: String
+Resources:
+  # Cloudformation does not support adding multiple certificats in Beanstalk
+  # with aws:elbv2:listener:443 -> SSLCertificateArns therefore we configure
+  # the Beanstalk ELB with a CFN resource in a separate step
+  SSLCertificate:
```
Comment (zaro0508) [inline templates/app-extra.yaml]: this isn't a certificate but a setting on a ALB correct?  maybe give it a name like `AlbHttpsListenerCertificate`?

**Review by zaro0508 — CHANGES_REQUESTED** [review]
would you mind creating the sceptre files (in config/develop and config/prod) that will run these templates?

File: templates/app-extra.yaml (hunk 1 above)
Comment (xschildw) [reply templates/app-extra.yaml:68]: I think it's fine. They are on different listeners.

File: templates/app-extra.yaml (hunk 2 above)
Comment (xschildw) [reply templates/app-extra.yaml]: Agree.

**Comment by xschildw** [comment]
I think I need 2 PRs: one to setup the the templates and setup ssl cert (added config), then the other one to add the config for app_extra (needs the cert created in first one). Would have been cleaner to have them logically separated (shared + shared config, then app_extra and app_extra config), I can tear down and redo if you prefer.

File: config/develop/shared.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+template_path: shared.yaml
```
Comment (zaro0508) [inline config/develop/shared.yaml]: need a `config/prod/shared.yaml` as well?

**Review by zaro0508 — CHANGES_REQUESTED** [review]
you only need one PR if you specify dependencies.  The config/develop/app-extra.yaml should depend on "develop/shared.yaml".  Please add a config/develop/app-extra.yaml and one for prod as well.

File: config/prod/shared.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+template_path: shared.yaml
```
Comment (zaro0508) [inline config/prod/shared.yaml]: ohh, i forgot we don't want to deploy this multiple times.  what i normally do is create a separate folder called "common",  there an example in agora project, https://github.com/Sage-Bionetworks/Agora-infra/tree/develop/config/common  and  then change travis to run only once, https://github.com/Sage-Bionetworks/Agora-infra/blob/develop/.travis.yml#L28

please follow that pattern.

File: config/develop/synapse-login-scipooldev-extra.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+template_path: app-extra.yaml
+stack_name: synapse-login-scipooldev-extra
+dependencies:
+  - develop/shared.yaml
```
Comment (zaro0508) [inline config/develop/synapse-login-scipooldev-extra.yaml]: should be 
```- common/shared.yaml```

see comment below

File: config/prod/synapse-login-scipoolprod-extra.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+template_path: app-extra.yaml
+stack_name: synapse-login-scipoolprod-extra
+dependencies:
+  - prod/shared.yaml
```
Comment (zaro0508) [inline config/prod/synapse-login-scipoolprod-extra.yaml]: should be 
```- common/shared.yaml```

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: config/develop/synapse-login-scipooldev-extra.yaml (hunk 5 above)
Comment (zaro0508) [reply config/develop/synapse-login-scipooldev-extra.yaml]: ignore this comment. i forgot it's running on two separate accounts

File: config/prod/shared.yaml (hunk 4 above)
Comment (zaro0508) [reply config/prod/shared.yaml]: please ignore my comment

File: config/prod/synapse-login-scipoolprod-extra.yaml (hunk 6 above)
Comment (zaro0508) [reply config/prod/synapse-login-scipoolprod-extra.yaml]: ignore.

**Review by zaro0508 — COMMENTED** [review]
now that i remember the resources are not `shared` how about changing the filename to `SageItCert.yaml`? and the stack names can be `synapse-login-scipooldev-sageit-cert`?

**Review by zaro0508 — APPROVED** [review]

**Pull request merged**